### PR TITLE
Add Prisma, db seed script, recipe APIs, frontend layout etc. 

### PR DESCRIPTION
**Pull request by blomqma**
This adds Prisma as a dependency as well
as a seed script to populate the database with
the sample data. This also adds the initial APIs
& frontend for listing recipes & favoriting them.
This also sets up the frontend layout, creating
the initial homepage contents.

### CODE REVIEW

**Comment by blomqma** [comment]
We need to enable the Prisma build plugin in the Netlify project for this repository in order to make sure the migrations work with the deployments. I don't think I have access to the Netlify project currently.

**Comment by blomqma** [comment]
How to test this locally:
* `npx prisma migrate dev` to run the migration (should also run the seed)
* `npx prisma db seed` to only run the seed data loading

**Comment by blomqma** [comment]
Here's a ref on how the Prisma seed works: https://www.prisma.io/docs/guides/migrate/seed-database#how-to-seed-your-database-in-prisma

File: prisma/migrations/migration_lock.toml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Please do not edit this file manually
+# It should be added in your version-control system (i.e. Git)
+provider = "postgresql"
```
Comment (blomqma) [inline prisma/migrations/migration_lock.toml:3]: This is generated by Prisma so the missing newline occurs on every new migration.

File: .gitignore [hunk 2]
```diff
@@ -25,7 +25,7 @@ yarn-debug.log*
 yarn-error.log*
 
 # local env files
-.env*.local
```
Comment (blomqma) [inline .gitignore:28]: Prisma doesn't support the `.env.local` file so I renamed it to just `.env`

**Comment by blomqma** [comment]
Here's a sneek peek on how this thing looks currently.

Screenshot 2023-06-08 at 17 10 16

Screenshot 2023-06-08 at 17 10 30

Screenshot 2023-06-08 at 17 10 41

File: src/components/Layout/Layout.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        <title>Recipe library powered by Aiven and Netlify</title>
+        <meta
+          name="description"
+          content="Recipe library built with Next.js, PostgreSQL, Redis, OpenSearch and Netlify"
+        />
+        <meta name="viewport" content="width=device-width, initial-scale=1" />
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+      <Navbar />
+      <main className={styles.main}>
```
Comment (dahlie) [inline src/components/Layout/Layout.tsx:24]: You *could* use `<Box.Flex component="main">` here as well. That way you could provide some of the custom styles as props instead. That way you'd be using, for example, same spacings as in Aquarium implementation.

File: src/components/Layout/Layout.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import React, { PropsWithChildren } from "react";
+import { Navbar } from "../Navbar/Navbar";
+import { Footer } from "../Footer/Footer";
+import Head from "next/head";
+import styles from "./styles.module.css";
+import { Alert } from "@aivenio/aquarium";
+import { useAppContext } from "@/context";
```
Comment (dahlie) [inline src/components/Layout/Layout.tsx:7]: NIT: Maybe we could include same sort linter config we have for Console for reordering and grouping imports.

File: src/pages/recipes/[recipeId].tsx [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+                  gap="5"
+                  style={{
+                    whiteSpace: "nowrap", // Not allowed by Aquarium - need to use inline styles.
+                  }}
+                >
+                  <Box.Flex
+                    flexDirection="column"
+                    justifyContent="space-between"
+                  >
+                    <Typography variant="body-default" fontWeight={600}>
```
Comment (dahlie) [inline src/pages/recipes/[recipeId].tsx:93]: There *should* not be need to use `<Typography>` like this. Would `<Typography.DefaultStrong>` work instead?

File: src/components/Layout/Layout.tsx (hunk 4 above)
Comment (blomqma) [reply src/components/Layout/Layout.tsx:7]: I'll create a follow-up action for setting up a Prettier config and using the Aiven ESlint config. Will do that either separately later or in this PR.

File: package.json [hunk 6]
```diff
@@ -4,20 +4,32 @@
   "private": true,
   "scripts": {
     "dev": "next dev",
-    "build": "next build",
+    "build": "prisma migrate reset --force && prisma generate && next build",
```
Comment (blomqma) [inline package.json:7]: This resets the database, runs the migrations and loads the sample data on every new deployment currently. Did it like that for now to test that seeding the database as part of the Netlify build works but for the live version we might not want to reset the database every time.

File: src/pages/api/recipes/index.ts [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+import type { NextApiRequest, NextApiResponse } from "next";
+import prisma, {
+  calculateFavoriteStats,
+  calculateTotalStats,
+} from "@/lib/prisma";
+import { createRedisInstance, getCachedDataFromRedis } from "@/lib/ioredis";
+import { REDIS_RECIPE_STATS_KEY } from "@/constants";
+import { ErrorResponse, RecipesResponse } from "@/types";
+
+const redis = createRedisInstance();
```
Comment (nctbao) [inline src/pages/api/recipes/index.ts:11]: should we create a common redis instance to be imported? I saw that we have two instances here and [here](https://github.com/aiven/nextjs-netlify/pull/2/files#diff-59ab858d61b08f82c4e6ce33fd4312182442be07a72d45ae30e3d95dd6508e21R11)

File: src/pages/recipes/index.tsx [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+        title="Statistics"
+        subtitle={
+          statistics?.recipeStatsCacheRetrievalTimeMs
+            ? `Recipe statistics cached in Aiven for Redis®, retrieved in ${statistics.recipeStatsCacheRetrievalTimeMs}ms (end-to-end with round
+          trip, results will vary depending on the region of your Aiven
+          for Redis® instance).`
+            : "Recipe statistics retrieved from PostgreSQL database. To get cached results using Aiven for Redis®, please follow the instructions to set up your Redis instance."
+        }
+      />
+      {data && (
```
Comment (nctbao) [inline src/pages/recipes/index.tsx:76]: I think here we could have a placeholder for loading state similar to how the table is currently. Reason is that the whole screen jumps quite a lot every time we click to another page.

File: src/components/Navbar/Navbar.tsx [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+import { Box, Button, Link } from "@aivenio/aquarium";
+
+export const Navbar: React.FC = () => (
+  <nav className={styles.navbar}>
+    <Link href="/">
+      <Image src="/logo.svg" width={150} height={50} alt="Aiven logo" />
+    </Link>
+    <Box marginLeft="5" display="flex" gap="5">
+      <Link href="/">Home</Link>
+      <Link href="/recipes">Recipes</Link>
```
Comment (nctbao) [inline src/components/Navbar/Navbar.tsx:13]: Should we have an active state for these links?

**Review by nctbao — COMMENTED** [review]
Awesome work! 💯  There are some improvements I noted down, if you want you can also create a ticket to address these comments later.

File: src/pages/recipes/[recipeId].tsx (hunk 5 above)
Comment (blomqma) [reply src/pages/recipes/[recipeId].tsx]: fixed everywhere

File: src/components/Navbar/Navbar.tsx (hunk 9 above)
Comment (blomqma) [reply src/components/Navbar/Navbar.tsx]: fixed

File: src/pages/recipes/index.tsx (hunk 8 above)
Comment (blomqma) [reply src/pages/recipes/index.tsx]: fixed

File: src/components/EmptyState/EmptyState.tsx [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+export const EmptyState: React.FC<Props> = ({ title }) => {
+  const router = useRouter();
+
+  return (
+    <Layout>
+      <Box width="full">
+        <AivenEmptyState
+          image="https://images.ctfassets.net/33bglmnnbgh7/7B19D37GjCHxatmfrb4MBw/e74b0f0c717a4cf3588e9f7ad15aa38e/aiven-observability-image-composition_1.png"
```
[the remark this answers is not in the document]
Comment (blomqma) [reply src/components/EmptyState/EmptyState.tsx]: fixed

**Review by nctbao — DISMISSED** [review]

**Pull request merged**